Shooter

zacharyCormack opened the pull request:
# ADDED SHOOTER MATH

added a shooter math function, which determines power to distance.
means driver need not intuitively understand fixed angle aim system.
also added empty joystick elements for shooter, as those were missing.

allanlzee committed:
Added Drivetrain, Spark Max Group Monitors, and Added Robot Constants.

allanlzee committed:
Added Wheel Diameter

allanlzee committed:
Added CANSparkMax Motor Monitors.

Added Motor Related Methods to the Drivetrain for Position, Velocity, and Turning.

allanlzee committed:
Added FollowTrajectory.java for Driving a Certain Distance.

Added Teloperated Drive Mode with Controller Bindings.

allanlzee committed:
Added DriveDistance.java.

This will drive the robot x inches forward, with PIDVADP Control.

allanlzee committed:
Configured Dashboard.

Added Tabs for Precision Drive, Ramping Rate, and Shutoff/Warning Temperatures.

allanlzee committed:
Added Collision Drive Command.

Will need to test collision threshold constant on the robot for better collision detection with fender.

allanlzee committed:
Updated CollisionDrive.java

allanlzee committed:
Added Intake Subsystem and Intake Command.

May need to edit if there are pneumatics/indexer tiggers.

allanlzee committed:
Updated RobotContainer.java

allanlzee committed:
Revert "Updated RobotContainer.java"

This reverts commit 6317d18648ebde6e990d93f1f5e6067d522e7587.

allanlzee committed:
Created Intake Subsystem Instance in Robot Container

allanlzee committed:
Added Logger Settings.

allanlzee committed:
Added Drive Distance Command.

allanlzee committed:
Modified .gitignore file.

allanlzee committed:
Added Arctos 6135 Libraries

allanlzee committed:
Added Arctos 6135 Libraries to Branch intake

allanlzee committed:
Added Arctos 6135 Libraries to Branch shooter

allanlzee committed:
Added Arctos 6135 Libraries to Branch dashboard

allanlzee committed:
Removed Unnecessary Imports

allanlzee committed:
Merge branch 'dashboard' into develop

allanlzee committed:
Removed Unnecessary Imports

allanlzee committed:
Merged Intake and Develop Branches.

allanlzee committed:
Removed Unnecessary Imports

allanlzee committed:
Merge branch 'shooter' into develop

allanlzee committed:
Added Shooter Subsystem.

Will need to confirm with shooter prototype before finalizing the subsystem.

allanlzee committed:
Changed REV Robotics Vendor Library.

allanlzee committed:
Added Motor Controllers, Encoders, and PID Controller to Shooter Subsystem.

allanlzee committed:
Added Monitor Group and Getter/Setter Functions for the Shooter Subsystem.

Update the shooter based on 2022 robot design.

allanlzee committed:
Added Shoot Command.

Shooting will require an indexer tigger system (possibly).

allanlzee committed:
Modified Shooter Subsystem.

allanlzee committed:
Updated Shoot Command.

allanlzee committed:
Added Shooter Subsystem Updates.

Updated Dashboard for the Shooter Subsystem.

zacharyCormack committed:
added basic shooter math function

TeamArctos6135 commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -35,252 +31,222 @@
+
+	}
+
+	private double shooterDistToSpeed(double dist, double endh) {
```
If this is a shooter-specific function, I think we should move this to the commands or subsystems folder as RobotContainer.java usually does not include any subsystem-specific functions.

zacharyCormack replied:
Ok, that's probably a good idea.

zacharyCormack replied:
I did that

TeamArctos6135 reviewed (changes requested):
To merge into the master branch, we should test the code on the robot (which has not been built yet). For now, I'll merge only the drivetrain into master and keep this on the shooter branch. Sound good?

allanlzee committed:
Added Monitor Group Widgets to RobotContainer.java.

These will monitor the motors used by the drivetrain and shooter subsystems.

allanlzee committed:
Merge branch 'shooter' of https://github.com/Arctos6135/frc-2022 into shooter

allanlzee committed:
Binding Shooter Related Commands to Xbox Buttons for the Operator Controller.

zacharyCormack committed:
added basic shooter steps

zacharyCormack committed:
moved shooter functionality to shooter subsystem

allanlzee committed:
Resolved Merge Conflicts.

Changed access modifiers of motors, controllers, and subsystems from private to default.

TeamArctos6135 committed:
Changed Shooter Command Bindings with the Controller Buttons.

Shooter is now operated with the operator controller instead of the driver controller.

TeamArctos6135 committed:
Added Constants for Xbox Button Bindings.

TeamArctos6135 committed:
Changed Gradle Distribution URL to 6.0.1 to successfully use ./gradlew build.

TeamArctos6135 committed:
Updated Intake Subsystem.

The intake subsystem uses one Spark MAX motor to spin the mecanum wheels to roll the ball into the shooter path (indexer tigger).
Another Spark MAX motor is used to rotate an intake arm with PIDF control so the wheels touch the ball.

TeamArctos6135 committed:
Added Shooter Feeder Subsystem.

The shooter feeder subsystem will roll balls towards the shooter. This should be implemented in all shooting-related commands.

TeamArctos6135 committed:
Added the ColorSensorV3 to the Shooter Feeder Subsystem.

The color sensor will detect balls and run the belts accordingly.
Note that REVLib.json causes ./gradlew build errors with the Spark MAX motors, so REVColorSensorV3.json will be used instead of REVLib.json.

TeamArctos6135 committed:
Updated Shooter Feeder Subsystem.

Added methods for changing the direction in which the feeder belts rotate.

TeamArctos6135 committed:
Added Shooter Feeder Command.

This command rolls balls up to the color sensor. Based on the color of the ball, the feeder mechanism can discard the ball or move it towards the shooter.

TeamArctos6135 committed:
Added Ball Counting Methods to the ShooterFeederSubsystem.

The ball count should be used when intaking, rolling, and shooting balls.

TeamArctos6135 committed:
Added SendableCANPIDController from FRC 2020.

This will allow PID controllers and their tuning values to be accessed via Shuffleboard.

zacharyCormack committed:
made Shooter.fire() actually do something.

TeamArctos6135 committed:
Merge branch 'shooter' of https://github.com/Arctos6135/frc-2022 into shooter

TeamArctos6135 committed:
Transferred Shooting Commands in RobotContainer.java to the Shoot.java command.

Added constructors for the custom shooter exception.

TeamArctos6135 committed:
Merged branches intake and shooter together.

Fixed Merge Conflicts between the two branches.
